server.js

```diff
@@ -18,9 +18,65 @@ const app = express()
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+// Mongoose model for Thought with properties message, hearts & createdAt
+const Thought = mongoose.model('Thought', {
+  message: {
+    type: String,
+    required: true,
+    minLength: 5,
+    maxLength: 140
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
+// Routes 
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Hello! Welcome to my Happy Thoughts API 💌')
+})
+
+// Endpoint to GET the 20 most recent thoughts, ordered by createdAt in descending order
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find().sort({ createdAt: 'desc' }).limit(20).exec()
+  res.json(thoughts)
+})
+
+// Endpoint to POST a new thought to the database
+app.post('/thoughts', async (req, res) => {
+  try {
+    // Success
+    const newThought = await new Thought(req.body).save()
+    res.status(200).json(newThought)
+    
+  // Bad request/error
+  } catch (err) {
+    res.status(400).json({ 
+      message: 'Error! Could not save happy thought.',
+      errors: err.errors
+    })
+  }
+})
+
+// Endpoint to POST likes on a certain thought, validated by id
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  try {
+    // Success
+    await Thought.updateOne({ _id: req.params.thoughtId }, { $inc: { hearts: 1 } })
+    res.status(200).json()
```

Comment:
@KarolCislowski I was just wondering if it's more appropriate to change this to be a 201 status message instead? :)